index.html

```diff
@@ -2258,6 +2249,15 @@ <h2 id="OfflineAudioContext">
               promise with <code>InvalidStateError</code>, and abort these
               steps.
               </li>
+              <li>Create a new <code>AudioBuffer</code>, with a number of
+              channels, length and sample rate equal respectively to the <code>
+                numberOfChannels</code>, <code>length</code> and
+                <code>sampleRate</code> values passed to this instance's
+                constructor in the <code>contextOptions</code> parameter.
+                Assign this buffer to <a><var>[[rendered buffer]]</var></a>. An
+                exception may be thrown during the constructor call, causing
+                the remaining steps in this process to be skipped.
+              </li>
```

Comment:
Shouldn't this say that startRendering will reject the promise?

I think this also means we should create the buffer after we've created a new promise so we have a promise to reject.

Reply:
In this instance, `startRendering()` throws an exception without even returning a Promise, which I think is appropriate. Other methods that return Promises can throw immediately, for instance if you pass invalid args to them...

Reply:
As a rather naive JS user, my expectation was that since `startRendering()` returns a promise, any errors or issues that occur should signal such things by rejecting the promise.

Passing invalid args to methods returning promises is different.  In essence, the method never actually had a chance to get started.  The error comes from parsing the args, the the guts of the method.

Reply:
Then it takes more work to handle three cases:

```js
try {
  oac.startRendering().then(resolutionHandler, rejectionHandler);
} catch (e) {
  errorHandler(e);
}
```

I believe it would be easier if all kinds of rejection/error can be handled by a single handler callback.

Reply:
OK, I'll have it return a rejected Promise in this case.